sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsAsyncClient.java

```diff
@@ -0,0 +1,228 @@
+    public Mono<LogsQueryBatchResultCollection> queryLogsBatch(String workspaceId, List<String> queries,
+                                                               QueryTimeSpan timeSpan) {
+        LogsQueryBatch logsQueryBatch = new LogsQueryBatch();
+        queries.forEach(query -> logsQueryBatch.addQuery(workspaceId, query, timeSpan));
+        return queryLogsBatchWithResponse(logsQueryBatch).map(Response::getValue);
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given batch of queries.
+     * @param logsQueryBatch {@link LogsQueryBatch} containing a batch of queries.
+     * @return A collection of query results corresponding to the input batch of queries.@return
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<LogsQueryBatchResultCollection>> queryLogsBatchWithResponse(LogsQueryBatch logsQueryBatch) {
+        return queryLogsBatchWithResponse(logsQueryBatch, Context.NONE);
+    }
+
+    Mono<Response<LogsQueryBatchResultCollection>> queryLogsBatchWithResponse(LogsQueryBatch logsQueryBatch, Context context) {
+        BatchRequest batchRequest = new BatchRequest();
+        AtomicInteger id = new AtomicInteger();
+
+        List<LogQueryRequest> requests = logsQueryBatch.getQueries()
+                .stream()
+                .map(query -> {
+                    QueryBody body = new QueryBody(query.getQuery())
+                            .setWorkspaces(query.getWorkspaceNames())
+                            .setAzureResourceIds(query.getAzureResourceIds())
+                            .setQualifiedNames(query.getQualifiedWorkspaceNames())
+                            .setWorkspaceIds(query.getWorkspaceIds());
+
+                    return new LogQueryRequest()
+                            .setId(String.valueOf(id.incrementAndGet()))
+                            .setBody(body)
+                            .setWorkspace(query.getWorkspaceId())
+                            .setPath("/query")
+                            .setMethod("POST");
+                })
+                .collect(Collectors.toList());
+        batchRequest.setRequests(requests);
+
+        return innerClient.getQueries().batchWithResponseAsync(batchRequest, context)
+                .map(this::convertToLogQueryBatchResult);
+    }
+
+    private Response<LogsQueryBatchResultCollection> convertToLogQueryBatchResult(Response<BatchResponse> response) {
+        List<LogsQueryBatchResult> batchResults = new ArrayList<>();
+        LogsQueryBatchResultCollection logsQueryBatchResultCollection = new LogsQueryBatchResultCollection(batchResults);
+
+        BatchResponse batchResponse = response.getValue();
+        for (LogQueryResponse singleQueryResponse : batchResponse.getResponses()) {
+            LogsQueryBatchResult logsQueryBatchResult = new LogsQueryBatchResult(singleQueryResponse.getId(),
+                    singleQueryResponse.getStatus(), getLogsQueryResult(singleQueryResponse.getBody()),
```

Comment:
>getLogsQueryResult [](http://example.com/codeflow?start=53&length=18)

Since getLogsQueryResult can in theory return null, batchResults could be a list with some null values. Not sure if that is possible though.

Reply:
Yes, that is acceptable because if `tables` is null, it means there is an error and the error field will be non-null.